lib/rspec/core/bisect/runner.rb

```diff
@@ -67,17 +74,22 @@ def run_locations(*capture_args)
           end
         end
 
+        # Consider just using Process.spawn since Open3 uses that.
```

Comment:
Did you try using `Process.spawn`?  Is there a reason you can't use that instead of mutating `ENV['SPEC_OPTS']`?